src/app/file/bmp_format.cpp

```diff
@@ -594,12 +923,16 @@ static int read_bitfields_image(FILE *f, Image *image, BITMAPINFOHEADER *infohea
       r = (buffer & rmask) >> rshift;
       g = (buffer & gmask) >> gshift;
       b = (buffer & bmask) >> bshift;
+      a = (buffer & amask) >> ashift;
 
-      r = rscale ? rscale(r): r;
-      g = gscale ? gscale(g): g;
-      b = bscale ? bscale(b): b;
+      r = (rbits == 8 ? r : scale_xxbits_to_8bits(rbits, r) );
+      g = (gbits == 8 ? g : scale_xxbits_to_8bits(gbits, g) );
+      b = (bbits == 8 ? b : scale_xxbits_to_8bits(bbits, b) );
+      a = (abits == 8 ? a : scale_xxbits_to_8bits(abits, a) );
```

Comment:
Please try to use the original scale_3bits_to_8bits() functions, it's a long history but I've to investigate a little more and write a blog post about this. (In the past I tweet about it: https://twitter.com/davidcapello/status/1242461399512756225 )

Reply:
Sorry, but `rbits / gbits / bbits / abits` can be any number between 0 and 32. So we need a function to process all the possible values.
Anyway we can include some particular cases with a `switch/case` inside of `scale_xxbits_to8bits()` function. What do you think?

Reply:
We can leave the switch/case version and see how to solve this in the future 👍